src/test/java/com/microsoft/jenkins/azuread/ObjId2FullSidMapTest.java

```diff
@@ -29,4 +33,49 @@ public void testPutAndGet() {
         Assert.assertEquals(FULL_SID_1, map.getOrOriginal(ObjId2FullSidMap.generateFullSid(NAME_1, OBJECT_ID_1)));
         Assert.assertEquals("some string", map.getOrOriginal("some string"));
     }
+
+    @Test
+    public void testExtractObjectId() {
+        Assert.assertNull(ObjId2FullSidMap.extractObjectId(""));
+        Assert.assertNull(ObjId2FullSidMap.extractObjectId("some string"));
+        Assert.assertNull(ObjId2FullSidMap.extractObjectId("email (id) "));
+        Assert.assertNull(ObjId2FullSidMap.extractObjectId("email (id"));
+        Assert.assertNull(ObjId2FullSidMap.extractObjectId("email id)"));
+        Assert.assertNull(ObjId2FullSidMap.extractObjectId("email(id)"));
+        Assert.assertEquals("id", ObjId2FullSidMap.extractObjectId("email (id)"));
+        Assert.assertEquals("", ObjId2FullSidMap.extractObjectId("email ()"));
+        Assert.assertEquals("id", ObjId2FullSidMap.extractObjectId(" (id)"));
+    }
+
+    @Ignore
+    @Test
+    public void testExtractObjectIdPerformance() throws Exception {
```

Comment:
FWIW JMH benchmarks are supported with something like this:
https://github.com/jenkinsci/junit-sql-storage-plugin/blob/master/src/test/java/io/jenkins/plugins/junit/storage/database/benchmarks/TrendGraphBenchmark.java

in case you are interested in converting but probably not worth it.

unsure if its worth committing this but certainly useful to provide it demonstrating the change.